MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,77 @@
+# Migration Guide
+
+## Upgrading from v0.46.0 → v0.47.0
+
+There are deprecations in this update. Please ensure you read this guide thoroughly and prepare your potential 
+automated workflows before upgrading.
+
+### Deprecations
+
+- [User Roles](#user-roles)
+
+### User Roles
```

Comment:
Assuming this is the only notable migration component in this h2, we don't need to establish such a rigid hierarchy.

```suggestion
### User Roles Deprecation
```

Reply:
I'd keep it like this for now and change it if we release the next version without the role_permission deprecation. Wdyt? Otherwise that will have it's own h3 element as well.